src/noshow/dashboard/connection.py

```diff
@@ -58,28 +58,49 @@ def get_patient_list(_session: Session, date_input: date) -> List[str]:
     List[str]
         List of unique patient ids, sorted by prediction
     """
-    # Calculate the threshold date for the last call
-    threshold_date = date.today() - timedelta(days=round(MUTE_PERIOD * 30.5))
 
-    call_list = _session.execute(
+    call_list_query = (
         select(
             ApiPrediction.patient_id,
             ApiPatient.treatment_group,
+            ApiPrediction.clinic_name,
             func.max(ApiPrediction.prediction),
         )
         .outerjoin(ApiPrediction.patient_relation)
         .where(ApiPrediction.start_time.cast(Date) == date_input)
         .where(ApiPrediction.active)
-        .where(
-            (ApiPatient.last_call_date.is_(None))
-            | (ApiPatient.last_call_date <= cast(threshold_date, Date))
-            | (ApiPatient.last_call_date == date.today())
-        )
         .where(ApiPatient.treatment_group >= 1)
         .where((ApiPatient.opt_out.is_(None)) | (ApiPatient.opt_out == 0))
-        .group_by(ApiPrediction.patient_id, ApiPatient.treatment_group)
+        .group_by(
+            ApiPrediction.patient_id,
+            ApiPatient.treatment_group,
+            ApiPrediction.clinic_name,
+        )
         .order_by(ApiPatient.treatment_group, func.max(ApiPrediction.prediction).desc())
-    ).all()
+    )
+
+    call_list = _session.execute(call_list_query).all()
+    mute_list = _get_mute_list(_session)
+
+    patient_dict = {}
+    for entry in call_list:
```

Comment:
This is not very efficient in Python, loops are slow and comparing tuples can also take some time. I think it's more efficient to do the filtering directly in the query above. 

A solution would be to do a left join of the tables above with ApiCallResponse and then in the 'where' clause check if the timestamp from the callresponse is either before the threshold date or if it is None (in case the patient was never called before)